src/FffGcodeWriter.cpp

```diff
@@ -371,6 +373,69 @@ void FffGcodeWriter::setInfillAndSkinAngles(SliceMeshStorage& mesh)
     }
 }
 
+void FffGcodeWriter::setSupportAngles(SliceDataStorage& storage)
+{
+    const Settings& mesh_group_settings = Application::getInstance().current_slice->scene.current_mesh_group->settings;
+
+    const size_t extruder_nr = mesh_group_settings.get<ExtruderTrain&>("support_infill_extruder_nr").extruder_nr;
+    const ExtruderTrain& support_infill_extruder = Application::getInstance().current_slice->scene.extruders[extruder_nr];
+    storage.support.support_infill_angles = support_infill_extruder.settings.get<std::vector<AngleDegrees>>("support_infill_angles");
+    if (storage.support.support_infill_angles.empty())
+    {
+        storage.support.support_infill_angles.push_back(0);
+    }
+
+    const size_t extruder_nr_layer_0 = mesh_group_settings.get<ExtruderTrain&>("support_extruder_nr_layer_0").extruder_nr;
+    const ExtruderTrain& support_extruder_nr_layer_0 = Application::getInstance().current_slice->scene.extruders[extruder_nr_layer_0];
+    storage.support.support_infill_angles_layer_0 = support_extruder_nr_layer_0.settings.get<std::vector<AngleDegrees>>("support_infill_angles");
+    if (storage.support.support_infill_angles_layer_0.empty())
+    {
+        storage.support.support_infill_angles_layer_0.push_back(0);
+    }
+
+    auto getInterfaceAngles = [&storage](const ExtruderTrain& extruder, const std::string& interface_angles_setting, const EFillMethod pattern, const std::string& interface_height_setting)
+    {
+        std::vector<AngleDegrees> angles = extruder.settings.get<std::vector<AngleDegrees>>(interface_angles_setting);
+        if (angles.empty())
+        {
+            if (pattern == EFillMethod::CONCENTRIC)
+            {
+                angles.push_back(0); //Concentric has no rotation.
+            }
+            else if (pattern == EFillMethod::TRIANGLES)
+            {
+                angles.push_back(90); //Triangular support interface shouldn't alternate every layer.
+            }
+            else
+            {
+                for (const SliceMeshStorage& mesh : storage.meshes)
+                {
+                    if (mesh.settings.get<coord_t>(interface_height_setting) >= 2 * Application::getInstance().current_slice->scene.current_mesh_group->settings.get<coord_t>("layer_height"))
+                    {
+                        //Some roofs are quite thick.
+                        //Alternate between the two kinds of diagonal: / and \ .
+                        angles.push_back(45);
+                        angles.push_back(135);
```

Comment:
I'm not sure about what I'm saying here, but I hope that this won't alternate directions every layer for some alternative patterns, such as Gyroid. Note to self: Test this well.

Reply:
Here the previous default behavior of `supportInterfaceFillAngle()` was preserved (if "support_roof_angles" or "support_bottom_angles" were left empty by user)

Reply:
Here the previous default behavior of `supportInterfaceFillAngle()` was preserved (if "support_roof_angles" or "support_bottom_angles" were left empty by user)